dask/tests/test_distributed.py

```diff
@@ -398,3 +335,93 @@ async def test_annotation_pack_unpack(c, s, a, b):
     unpacked_hlg = HighLevelGraph.__dask_distributed_unpack__(packed_hlg, annotations)
     annotations = unpacked_hlg["annotations"]
     assert annotations == {"workers": {"n": ("alice",)}}
+
+
+class SchedulerImportCheck(SchedulerPlugin):
+    """Plugin to help record which modules are imported on the scheduler"""
+
+    def __init__(self, pattern):
+        self.pattern = pattern
+
+    async def start(self, scheduler):
+        # Record the modules that have been imported when the scheduler starts
+        self.start_modules = set()
+        for mod in set(sys.modules):
```

Comment:
Doesn't seem to work without the `set` (or with `mod in sys.modules.keys()`).  My understanding is that, since we are modifying `sys.modules`,  we may need the `set` op to make a copy?

Reply:
Ah, my bad -- I didn't think about that